### PR TITLE
Ethereum Key-Type Signing

### PR DESCRIPTION
**Pull request by austinchandra**
Use Ethereum signing for `coinType = 60` for Evmos chain.

See: https://tharsis.notion.site/Keplr-Integration-PR-Docs-c1b06aa6a65b48e985f0ec468a1b0251

### CODE REVIEW

**Comment by Thunnini** [comment]
Can you provide more details on what the goal of the implementation is? is it just `MsgEthereumTx`?

**Comment by khoslaventures** [comment]
@Thunnini Hey, let me break it down into two categories.

Here's what we have to do by next week:
- Support the ethereum path for the key through the keyring (we are going to stop supporting the standard cosmos key path on mainnet because of EVM equivalence)
- Support `MsgEthereumTx` and Ethereum/Web3 signing with ethers.js or other signing library
- Support ledger bindings for Ethereum app to sign Ethereum/Web3 transactions
- Support connecting a wallet to a web3/EVM web page with Keplr

Stuff we will do later:
- Show bech32 and hex addresses in one UI
- Connect to the Ethereum/EVM RPC to show ERC20 balances and get events
- Inject Web3/Ethers.js into the webpage so existing EVM apps can work

**Comment by khoslaventures** [comment]
@Thunnini can you let us know if we're hitting the right spots for the current PR? We are waiting since it will define how the rest of the code is written.

**Comment by austinchandra** [comment]
@khoslaventures got most of it, so I'll keep it brief. This PR is primarily to create an endpoint by which a dApp can sign Ethereum transactions on the Evmos full-node via Keplr, similar to how `signDirect` works for Cosmos tx. In short, we allow Evmos-compatible websites to sign on Evmos.

The code is still work-in-progress, but we need your thoughts, @Thunnini, on architecture. And please let us know if anything here is unclear.

File: packages/background/src/keyring/messages.ts [hunk 1]
```diff
@@ -662,6 +662,79 @@ export class RequestSignDirectMsg extends Message<{
...
+  constructor(
+    public readonly chainId: string,
+    public readonly signer: string,
+    public readonly signDoc: {
+      bodyBytes?: Uint8Array | null;
+      authInfoBytes?: Uint8Array | null;
+      chainId?: string | null;
+      accountNumber?: string | null;
+    },
+    public readonly signOptions: KeplrSignOptions = {}
```
Comment (Thunnini) [inline packages/background/src/keyring/messages.ts:687]: It appears this is the same as `RequestSignDirectMsg`, what do you think of just adding `public readonly forEthermint?: boolean` to `RequestSignDirectMsg`?

File: packages/provider/src/core.ts [hunk 2]
```diff
@@ -188,6 +189,54 @@ export class Keplr implements IKeplr {
...
+    signer: string,
+    signDoc: {
+      bodyBytes?: Uint8Array | null;
+      authInfoBytes?: Uint8Array | null;
+      chainId?: string | null;
+      accountNumber?: Long | null;
+    },
+    signOptions: KeplrSignOptions = {}
+  ): Promise<DirectSignResponse> {
+    // How to access mnemonic to re-generate Eth key...
```
Comment (Thunnini) [inline packages/provider/src/core.ts:203]: this part runs in the content script, and there could be some potential security risk to handling sensitive data on the content script. You shouldn't create a msg that returns a mnemonic. This should only be processed in the background.

Comment (austinchandra) [reply packages/provider/src/core.ts]: Yeah, the mnemonic is only accessed within the keyring file, it shouldn't be exposed within the content script right now

File: packages/background/src/keyring/messages.ts (hunk 1 above)
Comment (austinchandra) [reply packages/background/src/keyring/messages.ts]: Just pushed a commit with this refactor

**Comment by austinchandra** [comment]
Hey @Thunnini, we've updated the PR with Cosmos signing using Ethereum keys, and are ready for review! Please take a look when you are available. Docs are here: https://tharsis.notion.site/Keplr-Integration-PR-Docs-c1b06aa6a65b48e985f0ec468a1b0251.

**Pull request merged**